sos/plugins/pulp.py

```diff
@@ -0,0 +1,65 @@
+# Copyright (C) 2018 Red Hat, Inc., Jake Hunsaker <jhunsake@redhat.com>
+
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.plugins import Plugin, RedHatPlugin
+from pipes import quote
+
+
+class Pulp(Plugin, RedHatPlugin):
+    """Pulp platform"""
+
+    plugin_name = "pulp"
+    packages = ("pulp-server", "pulp-katello")
+
+    def setup(self):
+        self.add_copy_spec([
+            "/etc/pulp/*.conf",
+            "/etc/pulp/server/plugins.conf.d/",
+            "/etc/default/pulp*",
+            "/var/log/httpd/pulp-http.log*",
+            "/var/log/httpd/pulp-https.log*",
+            "/var/log/httpd/pulp-http_access_ssl.log*",
+            "/var/log/httpd/pulp-https_access_ssl.log*",
+            "/var/log/httpd/pulp-http_error_ssl.log*",
+            "/var/log/httpd/pulp-https_error_ssl.log*"
+        ])
+
+        mtasks = self.build_mongo_cmd(
+            '\"DBQuery.shellBatchSize=${FOREMAN_DEBUG_MONGOTASKS:-200};; '
+            'db.task_status.find().sort({finish_time: -1})'
+            '.pretty().shellPrint()\"'
+        )
+
+        mres = self.build_mongo_cmd(
+            '\"db.reserved_resources.find().pretty().shellPrint()\"'
```

Comment:
Could we increase batch size (`DBQuery.shellBatchSize=..` like above) to 100 here? Default 20 can be too low for some cases.

Reply:
I havent tested this directly in sosreport, but in `bash`, `${FOREMAN_DEBUG_MONGOTASKS:-200}` expands to `${FOREMAN_DEBUG_MONGOTASKS}` if and only if the variable is defined, and expands to default `200` otherwise. Minus sign is part of `:-` operator here.

Reply:
Oh, derp. That should have been obvious to me.

I'll work on adding this as a plugin option, defaulting to 200.